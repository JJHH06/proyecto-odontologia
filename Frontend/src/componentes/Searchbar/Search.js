/* eslint-disable prefer-template */
import React, { useState } from 'react';
import './search.scss';
import Logotipo from '../../assets/BLANCO_Sin_Fondo.png';
import axios from 'axios';


// objeto de la calculadora
function Search() {
    const [pacientesBusqueda, setPacientesBusqueda] = useState("");
    const [pacientesEncontrados, setPacientesEncontrados] = useState([]);

    const handleSearch = (e) => {
        var data = JSON.stringify({
            "busqueda": pacientesBusqueda
          });
          
          var config = {
            method: 'post',
            url: 'http://localhost:5000/api/paciente/searchPaciente',
            headers: { 
              'Content-Type': 'application/json'
            },
            data : data
          };
          
          axios(config)
          .then(function (response) {
            console.log(JSON.stringify(response.data));
            setPacientesEncontrados(response.data)
          })
          .catch(function (error) {
            console.log(error);
          });
    }


    return (
        <div className = "container">
           
            <div className = "row">
                <div className = "col-10 input-group input-search-patient">
                <input type="text" className="form-control" placeholder="Nombre del paciente" aria-label="Recipient's username with two button addons" onChange={e =>setPacientesBusqueda(e.target.value)}/>
                
                <button type="submit" onClick = {handleSearch} className="btn btn-secondary btn-lg">Buscar</button>
                </div>
            </div>
            
            <hr className="featurette-divider search-separator"/>
            <div className = "row header-busqueda">
                <div className="col">
                    <b>Nombre</b>
                </div>
                <div className="col">
                    <b>Edad</b>

                </div>

                <div className="col">
                    <b>Telefono</b>

                </div>
            </div>
            {
                pacientesEncontrados.map((paciente,index) =>(
                    <div className = "row paciente-busqueda">
                <div className="col">
                        {paciente.nombre}
                </div>
                <div className="col">
                {paciente.edad}

                </div>

                <div className="col">
                {paciente.telefono_celular}
                </div>
                    </div>
                ))
            }

        </div>
    );
}

export default Search;
