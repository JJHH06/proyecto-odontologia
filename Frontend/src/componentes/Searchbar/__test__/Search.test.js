import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {render, cleanup, screen, fireEvent, waitFor} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Search from '../Search';




afterEach(cleanup);
it("Renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Search/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("Renders Search Correctly",() =>{
    const {getByTestId}= render(<Search/>)
    expect(getByTestId('search-patient')).toHaveTextContent("Buscar")
});

it('Search results are displayed', async () => {
    render(<Search />);
    expect(screen.getByTestId('cargando-info')).toBeInTheDocument();
});

it("Matches snapshot",() =>{
    const tree = renderer.create(<Search/>).toJSON();
    expect(tree).toMatchSnapshot();
})




