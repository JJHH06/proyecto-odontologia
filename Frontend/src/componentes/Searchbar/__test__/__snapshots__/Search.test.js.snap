// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="container"
  data-testid="search-patient"
>
  <div
    className="row"
  >
    <div
      className="col-10 input-group input-search-patient"
    >
      <input
        aria-label="Recipient's username with two button addons"
        className="form-control"
        onChange={[Function]}
        placeholder="Nombre del paciente"
        type="text"
      />
      <button
        className="btn btn-secondary btn-lg"
        data-testid="search-button"
        onClick={[Function]}
        type="submit"
      >
        Buscar
      </button>
    </div>
  </div>
  <hr
    className="featurette-divider search-separator"
  />
  <div
    className="row header-busqueda"
  >
    <div
      className="col"
    >
      <b>
        Nombre
      </b>
    </div>
    <div
      className="col"
    >
      <b>
        Edad
      </b>
    </div>
    <div
      className="col"
    >
      <b>
        Telefono
      </b>
    </div>
  </div>
  <span
    data-testid="cargando-info"
  />
</div>
`;
